runtime/src/bank.rs

```diff
@@ -7266,8 +7264,10 @@ pub(crate) mod tests {
         }
     }
 
-    fn map_to_test_bad_range() -> AccountMap<Pubkey, i8> {
-        let mut map: AccountMap<Pubkey, i8> = AccountMap::new();
+    fn map_to_test_bad_range() -> std::collections::BTreeMap<Pubkey, i8> {
+        // It looks like this is used to test both bank range functions as well as range errors from BTreeMap.
+        // We will likely refactor AccountsIndex to make a higher level fn range() that mimics the previous single BTreeMap::range.
+        let mut map: std::collections::BTreeMap<Pubkey, i8> = std::collections::BTreeMap::new();
```

Comment:
@ryoqun, AccountsIndex already doesn't work quite like this since we are now binning the index by pubkeys. Now that we plan to no longer use btree, that makes this code even less applicable. I could add a fn on AccountsIndex that returns a range, but I think references are going to get difficult to deal with since individual bins are behind their own rwlocks. If the purpose of this code is to verify that the ranges are being created correctly, then using a btree here is fine. If the purpose is to do more integration testing and test that the AccountsIndex is checking ranges correctly, then we have to rework this. Thoughts?

Reply:
Actually, see: #19225

Reply:
@jeffwashington this is only used to unit-test `pubkey_range_from_partition`. #19225 seems fine to merge in that regard. :)

Reply:
I reworked this separately in #19225.